sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/connectionstring/ConnectionStringSegments.java

```diff
@@ -8,6 +8,9 @@
  */
 final class ConnectionStringSegments {
 
+    private ConnectionStringSegments() {
```

Comment:
I am wondering if we could use Interface here, instead of declaring a private ctor. But that way, may bring more changes in other places.